gh-119740: Remove deprecated trunc delegation (#119743)
python/cpython: Doc/whatsnew/3.14.rst

  It had previously raised a :exc:`DeprecationWarning` since Python 3.9. (Contributed
  by Jelle Zijlstra in :gh:`118767`.)

> * The :func:`int` built-in will no longer delegate to

Author: ```suggestion
* The :func:`int` built-in no longer delegates to
```

> * The :class:`int` built-in will no longer delegate to

Reviewer: I think that it should be either ```The :func:`int` built-in``` or ```The :class:`int` constructor```. Preferably the former, since we will get a link to the constructor.

Author: Done in https://github.com/python/cpython/pull/119743/commits/7492da8cb64922147227858e9ef5cd120632a8ba

  :meth:`~object.__trunc__`. Classes that want to support conversion to

>   :func:`int` must implement either :meth:`~object.__int__` or

Reviewer: ```suggestion
  :func:`!int` must implement either :meth:`~object.__int__` or
```
To avoid repeating the same link within a paragraph.

Reviewer: Wait. it should be
```suggestion
  :class:`int` must implement either :meth:`~object.__int__` or
```
or
```suggestion
  :class:`!int` must implement either :meth:`~object.__int__` or
```
or simply
```suggestion
  integer must implement either :meth:`~object.__int__` or
```
We refer now to `int` as a type, not a callable.

>   :meth:`~object.__index__`. (Contributed by Mark Dickinson in :gh:`119743`).

Reviewer 2: NIT (for consistency):

```suggestion
  :meth:`~object.__index__`. (Contributed by Mark Dickinson in :gh:`119743`.)
```
